### ISSUE TITLE
Plugin not working in Godot 4.3

### ISSUE THREAD

**Issue by schaetz** [TheDuckCow/godot-road-generator#191]
I have tested this plugin before like half a year ago and it was working just fine at the time. My Godot version back then was probably 4.2.

I just retried it in Godot 4.3 and I can't create roads. I have created a fresh project, downloaded the latest Godot 4.x version of the plugin and followed the instructions in the "Getting Started" guide. But all that I can see when adding road points is blue tiles:

![image](https://github.com/user-attachments/assets/cc57fe15-67c0-4f6e-b522-70daf4f1d625)

Maybe this is some trivial mistake on my part? I double-checked the guide and watched the tutorial video and can't see what I am doing wrong.

**Comment by TheDuckCow** [comment]
I'll take a look at this more later, but I'm actually in the middle of making the first true alpha release for Godot 4.x, and have been testing via 4.3. Not everything is working yet, but it's mostly there vis the open pr (take note of the pr description):https://github.com/TheDuckCow/godot-road-generator/pull/190If you happen to try this branch, let me know if it works any better. I do suggest starting in a fresh scene to test, and definitely use the Roads menu > refresh roads button if ever unsure. On Sep 25, 2024 1:11 PM, Stefan Schätz ***@***.***> wrote:
I have tested this plugin before like half a year ago and it was working just fine at the time. My Godot version back then was probably 4.2.
I just retried it in Godot 4.3 and I can't create roads. I have created a fresh project, downloaded the latest Godot 4.x version of the plugin and followed the instructions in the "Getting Started" guide. But all that I can see when adding road points is blue tiles:
image.png (view on web)
Maybe this is some trivial mistake on my part? I double-checked the guide and watched the tutorial video and can't see what I am doing wrong.

**Comment by TheDuckCow** [comment]
Hi @schaetz the godot4 branch has been updated now, and should be fully functional based on my own testing. Let me know if it's working now, again being sure to restart the editor / reload the project after putting the node plugin code into place. Press refresh roads if needed. Worst case scenario, the scene and its roadpoint connections just got borqed and would need to be recreated in a fresh scene (or by manually clearing and re-assigning each prior/next point init, if the paths got corrupted).

**Comment by schaetz** [comment]
Sorry for my late response! I have been busy last week.

Yup, I created a new project and it works fine now. :) Thanks for the fix!

-----

I might have encountered another bug. Not sure if it is already a known issue. When I selected all nodes of my road at once by drawing a rectangle with my cursor, Godot crashed. There were about 30 nodes, I think. Could be both a Godot issue or an issue with this plugin.

If you want I can create a new ticket for it, however, I have not been able to reproduce it, yet.
